### ISSUE TITLE
proto files nested files?

### ISSUE THREAD

**Issue by MihailsKuzmins** [grpc/grpc-dotnet#1447]
I would like to split my proto file as follows:
```proto
syntax = "proto3";

option csharp_namespace = "GrpcService2";

package greet;
import "pkg1/greet.proto";

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (greet_types.HelloRequest) returns (greet_types.HelloReply);
}
```

and

```proto
syntax = "proto3";

option csharp_namespace = "GrpcService2";

package greet_types;

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings.
message HelloReply {
  string message = 1;
}
```

So in the file system it looks as follows
![image](https://user-images.githubusercontent.com/47413092/136854964-b3af32bf-fed9-4fc8-a067-5c79b9fa1deb.png)

But when I build the project there is an error:
![image](https://user-images.githubusercontent.com/47413092/136855020-a34df672-71ec-4803-9c45-c434bd782826.png)

I have also prepared a repo with this issue (https://github.com/MihailsKuzmins/GrpcService2)
Basically it is the standard project, but only greet.proto is split.

So, is it possible to accomplish what I want (split generation rpc and models).
My goal: generate models in the base project; generate client / server in child projects.

If it is not possible I will simply change gRPC models with other models, but I would like to keep gRPC models (less duplication)
Any help is much appreciated 🙏

**Comment by MihailsKuzmins** [comment]
@onimor thank you for your pieces of advice. It does work when I setup it in two projects.
In my actual solution generation of proto files is scattered in three projects - so in my test repository I have mocked the same structure and I get the same error (link to the PR https://github.com/MihailsKuzmins/GrpcService2/pull/1)

1. Project "Protobuf.Base1" - implement basic types
2. Project "Protobuf.Base2" - implement IMediator (should not be in the first project)
3. Project "GrpcService2" - server of gRPC

Basically the first two projects compile successfully, but as soon as I compile the last project (server code) the build fails.
As far I as can see project references and paths to Protobuf (`<Protobuf>`) are added correctly.

Maybe you would have time to look at my repository (https://github.com/MihailsKuzmins/GrpcService2) ? Maybe I have overlooked something?
![image](https://user-images.githubusercontent.com/47413092/137009447-6f302d03-7ad8-4f00-8326-a865af5fc3bd.png)

**Comment by chwarr** [comment]
Is the `ProtoBuf` item's `ProtoRoot` or `AdditionalImportsDirs` metadata set so that the import statements search paths will find the .proto files you're trying to import?

**Comment by JamesNK** [comment]
You need to include the complete path. It's not relative.

```
import "Protos/pkg1/greet.proto";
```
